### PR TITLE
Make profile args take precedence over env

### LINKED ISSUE #823: Cannot use `--dev` and `--release` flags in run scripts
image

### PR DESCRIPTION
**Pull request by szymmis**
Fixes #823

### CODE REVIEW

File: scarb/src/bin/scarb/args.rs [hunk 1]
```diff
@@ -345,7 +346,7 @@ pub struct GitRefGroup {
 #[group(multiple = false)]
 pub struct ProfileSpec {
     /// Specify profile to use by name.
-    #[arg(short = 'P', long, env = "SCARB_PROFILE")]
+    #[arg(short = 'P', long)]
```
Comment (mkaput) [inline scarb/src/bin/scarb/args.rs:349]: There is a little problem here, we remove information about the environment variable from `--help` :/

Zrzut ekranu 2023-10-25 o 08 53 56

**Review by mkaput — CHANGES_REQUESTED** [review]
I wonder if it wouldn't be better to allow multiple arguments in `ProfileSpec` group and employ overriding logic (`--dev` takes over `--release` which takes over `--profile`). Because we essentialy also have this problem if the user sets `export SCARB_PROFILE=...` in their `.rc` file etc.

File: scarb/tests/profiles.rs [hunk 2]
```diff
@@ -240,6 +240,33 @@ fn cannot_choose_not_existing_profile() {
...
+        .assert()
+        .success();
+
+    Scarb::quick_snapbox()
+        .args(["run", "build-release"])
+        .current_dir(&t)
+        .assert()
+        .success();
+}
+
```
Comment (maciektr) [inline scarb/tests/profiles.rs:269]: Can we check which profile has been used? 
You can read it from Scarb metadata: https://docs.rs/scarb-metadata/latest/scarb_metadata/struct.Metadata.html#structfield.current_profile

Can we check if cli args take precedence over env var?

**Review by maciektr — APPROVED** [review]

**Review by mkaput — APPROVED** [review]
👏🏻

**Pull request merged**